### PR TITLE
autorestic: init at 1.2.0

### PR DESCRIPTION
**Pull request by renesat**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/backup/autorestic/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ lib, fetchFromGitHub, installShellFiles, buildGoModule }:
+
+buildGoModule rec {
+  name = "autorestic";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "cupcakearmy";
+    repo = "${name}";
```
Comment (legendofmiracles) [inline pkgs/tools/backup/autorestic/default.nix:9]:
```suggestion
    repo = name;
```

File: pkgs/tools/backup/autorestic/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+         --fish <($out/bin/autorestic completion fish) \
+         --zsh <($out/bin/autorestic completion zsh) \
+  '';
+
+  meta = with lib; {
+    description = "High level CLI utility for restic";
+    homepage = "https://github.com/cupcakearmy/autorestic";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ renesat ];
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (legendofmiracles) [inline pkgs/tools/backup/autorestic/default.nix:32]: Set by buildGoModule. Only set this unless it really doesn't support anything else. But i wouldn't know why it couldn't support FreeBSD.

Comment (renesat) [reply pkgs/tools/backup/autorestic/default.nix:32]: I do not add FreeBSD because in package repo install script using only Linux and Darwin and in restic package usings this platforms.

File: pkgs/tools/backup/autorestic/default.nix (hunk 1 above)
Comment (renesat) [reply pkgs/tools/backup/autorestic/default.nix:9]: Do you mean `repo = restic` ?

File: pkgs/tools/backup/autorestic/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{ lib, fetchFromGitHub, installShellFiles, buildGoModule }:
+
+buildGoModule rec {
+  name = "autorestic";
```
Comment (legendofmiracles) [inline pkgs/tools/backup/autorestic/default.nix:4]:
```suggestion
 pname = "autorestic";
```

**Comment by renesat** [comment]
@legendofmiracles, fix `name` -> `pname`

File: pkgs/tools/backup/autorestic/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    owner = "cupcakearmy";
+    repo = "${pname}";
+    rev = "v${version}";
+    sha256 = "yQgSJ0SQNWPMyrYn8rep+1b549HP8sOERh+kOiAK3+c=";
+  };
+
+  vendorSha256 = "7648gAguqeqLKFS9xRcx20wpSLb+ykZ7rOqR5PKe71o=";
+
+  subPackages = [ "." ];
+
```
Comment (figsoda) [inline pkgs/tools/backup/autorestic/default.nix:17]: this is the default
```suggestion
```

File: pkgs/tools/backup/autorestic/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
+    installShellCompletion --cmd autorestic \
+         --bash <($out/bin/autorestic completion bash) \
+         --fish <($out/bin/autorestic completion fish) \
+         --zsh <($out/bin/autorestic completion zsh) \
```
Comment (figsoda) [inline pkgs/tools/backup/autorestic/default.nix:24]:
```suggestion
         --zsh <($out/bin/autorestic completion zsh)
```

File: pkgs/tools/backup/autorestic/default.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  '';
+
+  meta = with lib; {
+    description = "High level CLI utility for restic";
+    homepage = "https://github.com/cupcakearmy/autorestic";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ renesat ];
+    platforms = platforms.linux ++ platforms.darwin;
+  };
+
```
Comment (figsoda) [inline pkgs/tools/backup/autorestic/default.nix:34]:
```suggestion
```

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -9182,6 +9182,12 @@
     githubId = 546296;
     name = "Eric Ren";
   };
+  renesat = {
```
Comment (figsoda) [inline maintainers/maintainer-list.nix:9185]: this should be in a separate commit: `maintainers: add renesat`

**Review by figsoda — CHANGES_REQUESTED** [review]
just some small nitpicks

File: pkgs/tools/backup/autorestic/default.nix (hunk 1 above)
Comment (figsoda) [reply pkgs/tools/backup/autorestic/default.nix]: `pname` is the same as `"${pname}"` here since it is already a string

**Comment by renesat** [comment]
@figsoda, fixed it

**Review by figsoda — APPROVED** [review]

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 138350` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>autorestic</li>
  </ul>
</details>
